Exiled.Events/Patches/Events/Player/ReservedSlot.cs

```diff
@@ -0,0 +1,111 @@
+    ///     Adds the <see cref="Player.ReservedSlot" /> event.
+    /// </summary>
+    [HarmonyPatch(typeof(ReservedSlot), "HasReservedSlot")]
+    internal static class ReservedSlotPatch
+    {
+        [HarmonyTranspiler]
+        private static IEnumerable<CodeInstruction> ReservedSlotPatchTranspiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
+
+            LocalBuilder jumpConditions = generator.DeclareLocal(typeof(ReservedSlotEventResult));
+
+            Label continueConditions = generator.DefineLabel();
+
+            Label allowUnconditional = generator.DefineLabel();
+            Label returnTrue = generator.DefineLabel();
+            Label returnFalse = generator.DefineLabel();
+
+
+            int offset = -1;
+            int index = newInstructions.FindLastIndex(
+                instruction => instruction.LoadsField(Field(typeof(PlayerCheckReservedSlotCancellationData), nameof(PlayerCheckReservedSlotCancellationData.HasReservedSlot)))) + offset;
+
+            newInstructions[index].WithLabels(continueConditions);
+
+            newInstructions.InsertRange(
+                index,
+                new[]
+                {
+                    new CodeInstruction(OpCodes.Ldarg_0).MoveLabelsFrom(newInstructions[index]),
+                    new(OpCodes.Ldarg_1),
+                    new(OpCodes.Newobj, GetDeclaredConstructors(typeof(ReservedSlotsCheckEventArgs))[0]),
+                    new(OpCodes.Dup),
+                    new(OpCodes.Call, Method(typeof(Player), nameof(Player.OnReservedSlot))),
+                    new(OpCodes.Callvirt, PropertyGetter(typeof(ReservedSlotsCheckEventArgs), nameof(ReservedSlotsCheckEventArgs.Result))),
+                    new(OpCodes.Stloc, jumpConditions.LocalIndex),
+
+                    // Let normal NW code proceed. UseBaseGameSystem - 0 -> Allow base game check
+                    new(OpCodes.Ldloc, jumpConditions.LocalIndex),
+                    new(OpCodes.Brfalse_S, continueConditions),
+
+                    // Allow use of reserved slots, returning true CanUseReservedSlots - 1 - return true
+                    new(OpCodes.Ldloc, jumpConditions.LocalIndex),
+                    new(OpCodes.Ldc_I4_1),
+                    new(OpCodes.Beq, returnTrue),
+
+                    // Reserved slot rejection - CannotUseReservedSlots - 2 - return false
+                    new(OpCodes.Ldloc, jumpConditions.LocalIndex),
+                    new(OpCodes.Ldc_I4_2),
+                    new(OpCodes.Beq, returnFalse),
+
+                    // Allow unconditional connection - AllowConnectionUnconditionally - 3 - return true with bypass to true
+                    new(OpCodes.Ldloc, jumpConditions.LocalIndex),
+                    new(OpCodes.Ldc_I4_3),
+                    new(OpCodes.Beq, allowUnconditional),
```

Comment:
Do we need to branch to the allowUnconditional label that is right below? We're going to execute the code below anyway.

Reply:
In this case, no - if there are future cases, yes. I can remove it if you would prefer

Reply:
Yeah no problem, I only wanted to know if that was the reason, it's fine then.